skins/cat17/src/app/tests/view_handlers/test_section_info.js

```diff
@@ -422,3 +423,146 @@ test( 'Opened longtext are shut', function ( t ) {
+
+	t.ok( container.addClass.withArgs( 'completed' ).calledOnce, 'data entered reflected in style' );
+	t.ok( icon.addClass.withArgs( 'icon-person' ).calledOnce, 'icon set per address type' );
+	t.ok( text.text.withArgs( 'Privatperson' ).calledOnce, 'address type text is set' );
+	t.equals( longText.html.args[ 0 ][ 0 ].toString(), '<span>', 'long text filled with custom mark-up' );
+	t.equals( longText.html.args[ 0 ][ 0 ].append.args[ 0 ][ 0 ].text.args[ 0 ][ 0 ].toString(), 'Herr Dr. test user', 'name set' );
+	t.equals( longText.html.args[ 0 ][ 0 ].append.args[ 1 ][ 0 ].text.args[ 0 ][ 0 ].toString(), 'demostr 4', 'street set' );
+	t.equals( longText.html.args[ 0 ][ 0 ].append.args[ 2 ][ 0 ].text.args[ 0 ][ 0 ].toString(), '10112 Bärlin', 'address set' );
+	t.equals( longText.html.args[ 0 ][ 0 ].append.args[ 3 ][ 0 ].text.args[ 0 ][ 0 ].toString(), 'Deutschland', 'country translated and set' );
+	t.equals( longText.html.args[ 0 ][ 0 ].append.args[ 4 ][ 0 ].text.args[ 0 ][ 0 ].toString(), 'me@you.com', 'email set' );
+
+	delete global.jQuery;
+
+	t.end();
+} );
+
+test( 'Donor type info for company indicated correctly', function ( t ) {
+	var container = createContainerElement(),
+		icon = createElement(),
+		text = createElement(),
+		longText = createElement(),
+		handler = objectAssign( Object.create( SectionInfo.DonorTypeSectionInfo ), {
+			container: container,
+
+			icon: icon,
+			text: text,
+			longText: longText,
+
+			valueIconMap: { person: 'icon-person', firma: 'icon-firma', anonym: 'icon-anonym' },
+			valueTextMap: { person: 'Privatperson', firma: 'Firma', anonym: 'anonym' },
+
+			countryNames: { DE: 'Deutschland', AT: 'Österreich' }
+		} );
+
+	global.jQuery = jQueryPseudoHtmlGenerator;
+
+	handler.update( 'firma', 'Frau', 'Prof.', 'state left', 'from private', 'ACME INC', 'acmestr 133b', '12331', 'Wien', 'AT', 'us@acme.com', { dataEntered: true, isValid: true } );
+
+	t.ok( container.addClass.withArgs( 'completed' ).calledOnce, 'data entered reflected in style' );
+	t.ok( icon.addClass.withArgs( 'icon-firma' ).calledOnce, 'icon set per address type' );
+	t.ok( text.text.withArgs( 'Firma' ).calledOnce, 'address type text is set' );
+	t.equals( longText.html.args[ 0 ][ 0 ].toString(), '<span>', 'long text filled with custom mark-up' );
+	t.equals( longText.html.args[ 0 ][ 0 ].append.args[ 0 ][ 0 ].text.args[ 0 ][ 0 ].toString(), 'ACME INC', 'name set' );
+	t.equals( longText.html.args[ 0 ][ 0 ].append.args[ 1 ][ 0 ].text.args[ 0 ][ 0 ].toString(), 'acmestr 133b', 'street set' );
+	t.equals( longText.html.args[ 0 ][ 0 ].append.args[ 2 ][ 0 ].text.args[ 0 ][ 0 ].toString(), '12331 Wien', 'address set' );
+	t.equals( longText.html.args[ 0 ][ 0 ].append.args[ 3 ][ 0 ].text.args[ 0 ][ 0 ].toString(), 'Österreich', 'country translated and set' );
+	t.equals( longText.html.args[ 0 ][ 0 ].append.args[ 4 ][ 0 ].text.args[ 0 ][ 0 ].toString(), 'us@acme.com', 'email set' );
```

Comment:
There is a bunch duplication between these tests which makes me think code should be extracted.

Another thing I noticed here is the magical strings that need to be the same. Extracting these to whatever we use for constants in this version of JS would be nice.

These observations and suggestion of extraction are what prompted me into writing https://github.com/wmde/FundraisingFrontend/pull/1172/files#r160011395